crates/sui-graphql-rpc/src/types/query.rs

```diff
@@ -194,15 +201,33 @@ impl Query {
             )
             .await
             .extend(),
-            None => Object::query(ctx.data_unchecked(), address, ObjectVersionKey::Latest)
+            None => {
+                // top-level queries will always provide the latest checkpoint_sequence_number as context
+                let latest_checkpoint = Checkpoint::query_latest(ctx.data_unchecked())
+                    .await
+                    .extend()?;
+                Object::query(
+                    ctx.data_unchecked(),
+                    address,
+                    ObjectVersionKey::LatestAt(latest_checkpoint.stored.sequence_number as u64),
+                )
                 .await
-                .extend(),
+                .extend()
+            }
         }
     }
 
     /// Look-up an Account by its SuiAddress.
-    async fn address(&self, address: SuiAddress) -> Option<Address> {
-        Some(Address { address })
+    async fn address(&self, ctx: &Context<'_>, address: SuiAddress) -> Result<Address> {
```

Comment:
The draft schema indicates this should be optional, so just wanted to mention to double check.